### PR TITLE
SELV3-562: Repository initialization and server definition

### PR DESCRIPTION
**Pull request by anawrotsoldevelo**
This PR contains:
SELV3-562: Configure backend repository
SELV3-543: Create new schema in db called DHIS2
SELV3-544: Add new table which stores server data
SELV3-545: Create endpoints for adding new server, updating it, deleting

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,258 +1,2 @@
...
-* **DATABASE_URL** - The JDBC url under which the database is accessible. Our services use `jdbc:postgresql://db:5432/open_lmis` for connecting to the PostgreSQL database running in a container.
-* **POSTGRES_USER** - The username of the database user that the services should use. This variable is also used by our PostgreSQL container to create a user.
-* **POSTGRES_PASSWORD** - The password of the database user that the services should use. This variable is also used by our PostgreSQL container to create a user.
-
-These variables are used by our builds in order to integrate with the [Transifex](https://www.transifex.com/) translation management system:
-
-* **TRANSIFEX_USER** - The username to use with Transifex for updating translations.
-* **TRANSIFEX_PASSWORD** - The password to use with Transifex for updating translations.
+# OpenLMIS Selv v3 DHIS2 Integration Service
+Service for integrating SELV3 OpenLMIS implementation with DHIS2.
```
Comment (NikolaLaskowska) [inline README.md:2]: It would be worth to add a note how the docker image can be created to use it locally with disto

**Review by sradziszewski — COMMENTED** [review]
Final note about org.openlmis.integration.dhis2.web.server.**ServerController**.java

Let's consider separating the data flow in the API into 3 layers:
- repositories
- service
- controller

and placing all the logic that doesn't concern exposing the functionality to the user (controller layer) and data fetching/saving (repository layer) to the **service** class.

[EXAMPLE](https://tom-collings.medium.com/controller-service-repository-16e29a4684e5)

File: src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.util.UUID;
+import org.openlmis.integration.dhis2.ServerDataBuilder;
+import org.openlmis.integration.dhis2.domain.server.Server;
+import org.openlmis.integration.dhis2.repository.server.ServerRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.repository.CrudRepository;
+
+public class ServerRepositoryIntegrationTest extends BaseCrudRepositoryIntegrationTest<Server> {
+  @Autowired
```
Comment (kmadej) [inline src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java:26]: add empty line before autowired

File: src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  private ServerRepository serverRepository;
+
+  @Override
+  CrudRepository<Server, UUID> getRepository() {
```
Comment (kmadej) [inline src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java:30]: add public

File: src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    return serverRepository;
+  }
+
+  @Override
+  Server generateInstance() {
```
Comment (kmadej) [inline src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java:35]: add public

File: src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java (hunk 3 above)
Comment (kmadej) [reply src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java:30]: any reason to use default ?

File: src/main/java/org/openlmis/integration/dhis2/domain/BaseEntity.java [hunk 5]
```diff
@@ -31,12 +31,12 @@
 @ToString
 public abstract class BaseEntity {
 
-  private static final String UUID_TYPE = "pg-uuid";
+  public static final String UUID_TYPE = "pg-uuid";
```
Comment (kmadej) [inline src/main/java/org/openlmis/integration/dhis2/domain/BaseEntity.java:34]: any reason ?

File: src/main/java/org/openlmis/integration/dhis2/dto/server/ServerDto.java [hunk 6]
```diff
@@ -13,33 +13,33 @@
...
 
 @Getter
 @Setter
 @NoArgsConstructor
 @AllArgsConstructor
 @EqualsAndHashCode(callSuper = true)
 @ToString(callSuper = true)
-public final class WidgetDto extends BaseDto implements Widget.Importer, Widget.Exporter {
+public final class ServerDto extends BaseDto implements Server.Importer, Server.Exporter {
   private String name;
```
Comment (kmadej) [inline src/main/java/org/openlmis/integration/dhis2/dto/server/ServerDto.java:34]: add empy line

**Review by kmadej — CHANGES_REQUESTED** [review]

File: src/main/java/org/openlmis/integration/dhis2/domain/server/Server.java [hunk 7]
```diff
@@ -66,23 +77,22 @@ public void updateFrom(Importer importer) {
...
   }
 
   public interface Importer extends BaseImporter {
 
     String getName();
 
-    String getCode();
+    String getUrl();
 
   }
```
[the remark this answers is not in the document]
Comment (anawrotsoldevelo) [reply src/main/java/org/openlmis/integration/dhis2/domain/server/Server.java:98]: Our checkstyle enforces method definition separation so I cannot change it. 
When it comes to separating data flow into 3 layers, if we decide to do so we should change template repository simultaneously. Separate ticket will also be needed for this. Anyways I think it is worth considering.

File: src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java (hunk 2 above)
Comment (anawrotsoldevelo) [reply src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java:26]: fixed

File: README.md (hunk 1 above)
Comment (anawrotsoldevelo) [reply README.md:2]: Added note about docker image creation in upcoming commit

File: src/main/java/org/openlmis/integration/dhis2/dto/server/ServerDto.java (hunk 6 above)
Comment (anawrotsoldevelo) [reply src/main/java/org/openlmis/integration/dhis2/dto/server/ServerDto.java:34]: fixed in whole project

File: src/main/java/org/openlmis/integration/dhis2/domain/BaseEntity.java (hunk 5 above)
Comment (anawrotsoldevelo) [reply src/main/java/org/openlmis/integration/dhis2/domain/BaseEntity.java:34]: fixed

File: src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java (hunk 4 above)
Comment (anawrotsoldevelo) [reply src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java:35]: fixed in whole project

File: src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java (hunk 3 above)
Comment (anawrotsoldevelo) [reply src/integration-test/java/org/openlmis/integration/dhis2/repository/ServerRepositoryIntegrationTest.java:30]: fixed

**Review by NikolaLaskowska — APPROVED** [review]

**Pull request merged**